chore: enable more ruff rules, fix

akx opened the pull request:
Some of these are manual fixes, some are automated. 

Would recommend manual review! Each commit should be reviewable in isolation, and there should be no back-and-forth between them.

akx commented:
Might be a good idea to get #340 merged first so Ruff linting is enforced in CI.

japsu commented on backend/access/cbac.py:
```diff
@@ -135,7 +136,7 @@ def is_graphql_allowed_for_model(
     user = info.context.user
     operation = "query"
-    extra = dict(slug=instance.slug) if hasattr(instance, "slug") else {}
+    extra = {"slug": instance.slug} if hasattr(instance, "slug") else {}
```
This is a personal preference and I will keep writing `dict(foo=5)` :) But no need to remove this from the PR.

akx replied:
Yeah, this got changed automagically somewhere else as well, and I reverted it then too. Will revert this too while rebasing, no biggie!

akx replied:
Also, the personal preference is noted by
```toml
ignore = [
  # ...
  "C408",  # unnecessary dict call (these are likely stylistic)
]
```
😌

japsu commented:
<img width="434" alt="image" src="https://github.com/con2/kompassi/assets/109397/8085351c-72e7-4c6a-9dc1-a11e23dac059">

404 on `/graphql` is a bit rough, no idea what's causing it but obv can't merge as-is @akx

japsu reviewed (changes requested):
need to fix 404 on `/graphql`

akx commented:
> 404 on `/graphql` is a bit rough, no idea what's causing it but obv can't merge as-is

weh, it was getting mounted at a wrong path due to a brainfart:

<img width="405" alt="Screenshot 2024-01-16 at 19 10 57" src="https://github.com/con2/kompassi/assets/58669/694d1397-9394-42db-bab9-1ac9db06aebd">

Rebased, fixed that, added a new commit that tests that the GraphQL API is mounted at the well-known path `/graphql`...
